addons/account/models/account_journal.py

```diff
@@ -402,6 +402,7 @@ def _inverse_type(self):
         if not self._context.get('account_journal_skip_alias_sync'):
             for record in self:
                 record._update_mail_alias()
+        self.filtered(lambda journal: journal.type == 'general').default_account_id = False
```

Comment:
```suggestion
        update_alias = not self._context.get('account_journal_skip_alias_sync')
        for journal in self:
            if update_alias:
                journal._update_mail_alias()
            # Since 'default_account_id' is not visible on MISC ('general') journals,
            # any existing value should be cleared to avoid undesired side effects.
            if journal.type == 'general':
                journal.default_account_id = False
```
"Slightly" more verbose, but it felt a bit weird to have both a `for` loop and a `filtered()` while both could have been done together.
Plus it allows to add come context in the comments :nerd_face: